sys/net/gnrc/network_layer/ipv6/nib/nib_ft.c

```diff
@@ -36,12 +36,22 @@ int gnrc_ipv6_nib_ft_get(const ipv6_addr_t *dst, gnrc_pktsnip_t *pkt,
 
 int gnrc_ipv6_nib_ft_add(const ipv6_addr_t *dst, unsigned dst_len,
                          const ipv6_addr_t *next_hop, unsigned iface,
-                         uint16_t ltime)
+                         uint32_t ltime)
 {
     int res = 0;
     bool is_default_route = ((dst == NULL) || (dst_len == 0) ||
                              ipv6_addr_is_unspecified(dst));
 
+    if (ltime < UINT32_MAX) {
+        /* UINT32_MAX means infinite lifetime.
```

Comment:
but the doc says

```suggestion
        /* 0 means infinite lifetime.
```

Reply:
But it means you can remove / modify the check:

```C
if (ltime > UINT32_MAX / MS_PER_SEC) {
    ltime_ms = UINT32_MAX;
} else {
     ltime_ms = ltime * MS_PER_SEC;
}
```

Otherwise `ltime_ms` will overflow if `ltime` is set to `UINT32_MAX - 1`